### DISCUSSION TITLE
V9 Release: when?

### DISCUSSION THREAD

**Discussion by frankvolkel** [directus/directus#2708] [Archive]
I've hit the same bug as detailed in https://github.com/directus/directus/issues/2585

The fix is coming in V9: when will V9 be released? 

Thanks for the awesome work :)

**Comment by rijkvanzanten** [comment]
Whenever it's ready!

On a more serious note, it's incredibly hard to give a super pinpointed date, as it's all open source and mostly community driven. That being said, we're aiming to release an alpha this week so we can get some initial community feedback 🙂

**Reply by frankvolkel** [reply]
Great. I'm seriously considering building on Directus for an upcoming project, but the bug is a huge blocker. 

Will the alpha be upgradable to later releases?

**Reply by rijkvanzanten** [reply]
Yeah definitely. It'll be a fairly standard alpha-beta-rc-prod release flow where all versions can be upgraded to the prod once that's released

**Reply by cupcakearmy** [reply]
@rijkvanzanten Is there something blocking the release? 
I'm waiting for https://github.com/directus/app/issues/1996, otherwise unfortunately we need to look somewhere else :/
Thanks for all the work

**Reply by benhaynes** [reply]
As an OSS product, with a team of only 4, working with about $75/month of donations, we can't really give hard timelines for releases. All I can say is that for this v9 major version we're focusing on stability... so we'll be pushing it when it's "ready". That said, we're about a week from Alpha and 2-3 weeks from Beta. We'll have a good estimate for an RC or production release once we're in beta.

**Reply by cupcakearmy** [reply]
Thanks! I know this is OSS :) no pressure, just wanted to hear what the current state is :)

**Reply by benhaynes** [reply]
Haha, $76/month! Thank you, @cupcakearmy !! ❤️

**Reply by cupcakearmy** [reply]
@benhaynes ist there an issue that I can track? I really don't want to bother you every-time I'm curious what the current state is ❤️  I'm so exited for the new release 😅

**Reply by benhaynes** [reply]
You can get the most up-to-date info on our Discord (we're switching our 2.5K community from Slack to Discord), or via Twitter:

https://discord.gg/directus
https://twitter.com/directus

Both of those will announce updates, with Discord updates being pretty much daily.

**Reply by rijkvanzanten** [reply]
Also keep an eye on https://github.com/directus/next @cupcakearmy 🙂

**Reply by niklasgrewe** [reply]
@rijkvanzanten I have two more questions: 

1. will it be possible to upgrade from v8 to v9 even when using Docker? 

2. Can I already test the Alpha in a docker container? Do you plan to release the official Docker image after prod release?

**Reply by rijkvanzanten** [reply]
> will it be possible to upgrade from v8 to v9 even when using Docker?

Yeah, the usage of Docker doesn't really matter in that regard. There are a couple breaking changes in the DB regarding permissions and interface options, so there will be some manual tweaking involved.

> Can I already test the Alpha in a docker container? Do you plan to release the official Docker image after prod release?

I haven't made a docker container yet, but seeing it's a fairly standard NodeJS based app, you should be able to follow something like https://nodejs.org/en/docs/guides/nodejs-docker-webapp/ to get up and running.

**Reply by cupcakearmy** [reply]
Wait... is the v9 completely rewritten in node??? @rijkvanzanten

**Reply by benhaynes** [reply]
Directus 9 has been completely rewritten in both PHP (Laravel) and Node.js... so yes. The node port is out in Alpha now on the `next` branch, Laravel is still in-work but on track for Alpha in a few weeks.

**Reply by cupcakearmy** [reply]
omg you guys are insane ❤️

**Comment by Zyles** [comment]
Wait what? There will be two code bases of the same project, PHP and JS?

**Reply by benhaynes** [reply]
Yup.
